lib/ar-find-in-batches-with-order.rb

```diff
@@ -30,27 +27,27 @@ def find_in_batches_with_order(options = {})
 
         break if records_size < batch_size
 
-        start = records.last.try(property_key)
+        next_start = records.last.try(property_key)
+        with_start_ids.clear if start != next_start
+        start = next_start
 
-        records = strict_mode ? (direction == :desc ? relation.where("#{sanitized_key} <= ?", start).to_a : relation.where("#{sanitized_key} >= ?", start).to_a) : (direction == :desc ? relation.where("#{sanitized_key} < ?", start).to_a : relation.where("#{sanitized_key} > ?", start).to_a)
+        records.each do |record|
+          if record.try(property_key) == start
+            with_start_ids << record.id
+          end
+        end
+
+        without_dups = relation.where.not(relation.klass.primary_key => with_start_ids)
```

Comment:
Above you assume `.id` for the records, but here you precisely extract the actual primary key column, so maybe apply the same above?

Reply:
Should we avoid even adding the `.not` if `start_ids` is empty?

Reply:
the id method actually uses primary_key:  https://apidock.com/rails/ActiveRecord/Base/id

with_start_ids always includes at least the key from records.last so it is never empty